lib/src/insertion_order_parser.dart

```diff
@@ -82,8 +82,18 @@ class InsertionOrderParser {
       ..budgetUnit = _createBudgetUnit(map['budgetUnit'])
       ..automationType = _createAutomationType(map['automationType']);
 
+    final now = DateTime.now();
     for (Map segmentMap in map['budgetSegments'] ?? []) {
-      budget.budgetSegments.add(_createBudgetSegment(segmentMap));
+      final segment = _createBudgetSegment(segmentMap);
```

Comment:
Moving this active segment calculation from population to parsing. Shall we modify the proto definition to just make budgetSegment optional instead of repeated since there is only gonna be one active segment at a time?

Reply:
yea +1
feel free to change in this pr if you want